### PR TITLE
Multiple token pairs

### PR DESCRIPTION
**Pull request by sashimi36**
This is to allow for adding pools with any possible token pairs

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -5,9 +5,8 @@ NEXT_PUBLIC_APP_VERSION=1.2.5
 
 # Asset Information
 NEXT_PUBLIC_CHAIN_INFO_URL=/chain_info.testnet.json
-NEXT_PUBLIC_TOKEN_LIST_URL=/token_list.testnet.json
+NEXT_PUBLIC_POOLS_LIST_URL=/pools_list.testnet.json
```
Comment (ben2x4) [inline .env:10]: love this, less json!

File: features/liquidity/components/BondLiquidityDialog.tsx [hunk 2]
```diff
@@ -26,23 +24,37 @@ import React, { useEffect, useRef, useState } from 'react'
...
 
-  const [{ myLiquidity } = {} as any] = usePoolLiquidity({ poolId })
-  const [stakedAmount] = useStakedTokenBalance({ poolId })
+  const { pool_assets, liquidity } = pool || {}
+  const [tokenA, tokenB] = pool_assets || []
 
   const maxLiquidityTokenAmount =
-    dialogState === 'stake' ? myLiquidity?.tokenAmount ?? 0 : stakedAmount ?? 0
+    dialogState === 'stake'
+      ? liquidity?.fluid.provided.tokenAmount ?? 0
```
Comment (ben2x4) [inline features/liquidity/components/BondLiquidityDialog.tsx]: can we change the word fluid? Maybe available is a better term? open to suggestions

File: features/liquidity/components/ManageLiquidityCard.tsx [hunk 3]
```diff
@@ -13,16 +12,16 @@ import {
...
-  | 'myLiquidityReserve'
-  | 'myStakedLiquidityReserve'
-  | 'tokenDollarValue'
-  | 'myStakedLiquidity'
-> & {
+type ManageLiquidityCardProps = {
+  stakedLiquidity: PoolState
+  providedLiquidity: PoolTokenValue
+  providedTotalLiquidity: PoolTokenValue
+  providedLiquidityReserve: ReserveType
```
Comment (ben2x4) [inline features/liquidity/components/ManageLiquidityCard.tsx:23]: what is the difference between provided liquidity and provided liquidity reserve?

File: features/liquidity/components/ManagePoolDialog/usePoolDialogController.tsx [hunk 4]
```diff
@@ -12,6 +9,7 @@ import {
   UpRightArrow,
   Valid,
 } from 'junoblocks'
+import { PoolEntityTypeWithLiquidity } from 'queries/useQueryPools'
```
Comment (ben2x4) [inline features/liquidity/components/ManagePoolDialog/usePoolDialogController.tsx:12]: I'm a little confused what PoolEntityType means? I think we can use a clear name?

File: features/liquidity/components/ManagePoolDialog/usePoolDialogController.tsx [hunk 5]
```diff
@@ -84,27 +81,28 @@ export const usePoolDialogController = ({
...
   const { isLoading, mutate: mutateAddLiquidity } = useMutateLiquidity({
+    pool,
     actionState,
     percentage,
     tokenA,
     tokenB,
     maxApplicableBalanceForTokenA,
     maxApplicableBalanceForTokenB,
-    myLiquidity,
+    providedLiquidity: liquidity?.fluid?.provided,
```
Comment (ben2x4) [inline features/liquidity/components/ManagePoolDialog/usePoolDialogController.tsx]: same comment about using the term fluid

File: features/liquidity/components/PoolCard.tsx [hunk 6]
```diff
@@ -42,28 +42,24 @@ export const PoolCard = ({
...
   const providedLiquidityDollarValueFormatted = hasProvidedLiquidity
-    ? formatToCompactDollarValue(
-        myLiquidity.dollarValue + stakedTokenBalanceDollarValue
-      )
+    ? formatToCompactDollarValue(providedTotalLiquidity.dollarValue)
     : 0
 
   const totalDollarValueLiquidityFormatted = dollarValueFormatterWithDecimals(
-    totalLiquidity.dollarValue,
+    fluidLiquidity.total.dollarValue,
```
Comment (ben2x4) [inline features/liquidity/components/PoolCard.tsx]: fluid :P

File: pages/pools/index.tsx [hunk 7]
```diff
@@ -69,18 +69,20 @@ export default function Pools() {
...
+                {myPools.map(
+                  ({ liquidity, pool_id, pool_assets: [tokenA, tokenB] }) => (
+                    <PoolCard
+                      key={pool_id}
+                      tokenASymbol={tokenA.symbol}
+                      poolId={pool_id}
+                      tokenBSymbol={tokenB.symbol}
+                      providedTotalLiquidity={liquidity.providedTotal}
+                      stakedLiquidity={liquidity.staked}
+                      fluidLiquidity={liquidity.fluid}
```
Comment (ben2x4) [inline pages/pools/index.tsx]: fluid

File: pages/pools/index.tsx [hunk 8]
```diff
@@ -102,41 +104,27 @@ export default function Pools() {
...
+            {allPools?.map(
+              ({ liquidity, pool_id, pool_assets: [tokenA, tokenB] }) => (
+                <PoolCard
+                  key={pool_id}
+                  tokenASymbol={tokenA.symbol}
+                  poolId={pool_id}
+                  tokenBSymbol={tokenB.symbol}
+                  providedTotalLiquidity={liquidity.providedTotal}
+                  stakedLiquidity={liquidity.staked}
+                  fluidLiquidity={liquidity.fluid}
```
Comment (ben2x4) [inline pages/pools/index.tsx]: fluid

**Review by ben2x4 — APPROVED** [review]
looks great, tested it with prod confg and will do more testing when merged. Just a few mino comments about variable names

File: features/liquidity/components/ManageLiquidityCard.tsx (hunk 3 above)
Comment (sashimi36) [reply features/liquidity/components/ManageLiquidityCard.tsx:23]: `providedLiquidityReserve` is an array with 2 numbers. One is for the first token pair in the pool and the other one is for the second token. `providedLiquidity` contains full lp tokenAmount the user provided in the pool and what it is in dollar value

File: features/liquidity/components/ManagePoolDialog/usePoolDialogController.tsx (hunk 4 above)
Comment (sashimi36) [reply features/liquidity/components/ManagePoolDialog/usePoolDialogController.tsx:12]: Open to suggestions but PoolEntity is an entity of a pool and the pool in this case is what we have in the pool list json

**Pull request merged**